2023-04-22 14:37:56,108 ERROR Internal Server Error: /news/ C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\utils\log.py (<class 'NameError'>, NameError("name 'sa' is not defined"), <traceback object at 0x0000020BAE322600>)
Traceback (most recent call last):
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\NewsPaper\news\views.py", line 24, in index
    logger.error(sa)
NameError: name 'sa' is not defined
2023-04-22 14:37:56,161 ERROR "GET /news/ HTTP/1.1" 500 74502 C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\servers\basehttp.py None
2023-04-22 15:25:21,365 ERROR Internal Server Error: /news/ C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\utils\log.py (<class 'django.template.exceptions.TemplateSyntaxError'>, TemplateSyntaxError("Invalid block tag on line 84: 'endblock'. Did you forget to register or load this tag?"), <traceback object at 0x00000252A38F8700>)
Traceback (most recent call last):
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 505, in parse
    compile_func = self.tags[command]
KeyError: 'endblock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\NewsPaper\news\views.py", line 27, in index
    return render(request, 'posts.html', context={'posts': posts})
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\shortcuts.py", line 24, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\backends\django.py", line 34, in get_template
    return Template(self.engine.get_template(template_name), self)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 507, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\template\base.py", line 568, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 84: 'endblock'. Did you forget to register or load this tag?
2023-04-22 15:25:21,461 ERROR "GET /news/ HTTP/1.1" 500 196771 C:\Users\Asus\PycharmProjects\Django\NewsPaper1\venv\lib\site-packages\django\core\servers\basehttp.py None
